### PR TITLE
fix for no Go source files on macOS

### PR DESCRIPTION
**Pull request by dims**
Make things a bit more easier on macOS
```
~/go/src/github.com/google/cadvisor
$ go list all
build k8s.io/utils/inotify: cannot load k8s.io/utils/inotify: no Go source files
```

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

### CODE REVIEW

File: inotify/inotify_others.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package inotify // import "k8s.io/utils/inotify"
+
+import (
+	"fmt"
+	"runtime"
+)
+
+// NewWatcher creates and returns a new inotify instance using inotify_init(2)
+func NewWatcher() (*Watcher, error) {
+	return nil, fmt.Errorf("watch not supported on %s", runtime.GOOS)
```
Comment (neolit123) [inline inotify/inotify_others.go]: possibly the error can shared across all functions.

```
var notSupportedError = fmt.Errorf("watch not supported on %s", runtime.GOOS)
...

func NewWatcher() (*Watcher, error) {
	return nil, notSupportedError 
}
...
```

Comment (dims) [reply inotify/inotify_others.go]: DONE!

**Comment by neolit123** [comment]
> Errors from golint:
inotify/inotify_others.go:26:5: error var notSupportedError should have name of the form errFoo

my bad, looks like it wants `errNotSupported` instead.

**Pull request merged**